snuba/web/query.py

```diff
@@ -195,48 +224,87 @@ def raw_query(
     return result
 
 
-def log_query_and_update_stats(
+def update_query_list_and_stats(
     request: Request,
     sql: str,
     timer: Timer,
     stats: MutableMapping[str, Any],
-    status: str,
+    query_list: MutableSequence[ClickhouseQueryMetadata],
     query_settings: Mapping[str, Any],
+    trace_id: str,
+    status: str,
 ) -> MutableMapping:
     """
     If query logging is enabled then logs details about the query and its status, as
     well as timing information.
     Also updates stats with any relevant information and returns the updated dict.
     """
-
     stats.update(query_settings)
+
+    query_list.append(
+        ClickhouseQueryMetadata(
+            request=request.body,
```

Comment:
I think the query AST would be more useful here. I would advise to write a concise formatter visitor so write the format you prefer, It can even be a dict.
Suggesting this because, as soon as I manage to support AST changes in the splitter and we stop using the old body, request.body would not be the source of truth anymore for what is being executed. The query object is that source of truth instead.

Reply:
To be clear, I don't think you need to write a new formatter within this PR. I would just remove the request.body from Clickhouse metadata and add it back together with a query formatter in a separate PR.

Reply:
Ok, removed it